### PR TITLE
Producer jenkins

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available:*
#68, #106 

*Description of changes:*
- Jenkins pipeline for producer SDK

Todo:
- Add design document

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:66]: Ditto

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:38]: To be less repetitive in handling credentials, environment variables, and error handling, you can create a wrapper function similar to what webrtc canary has:

https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/d163c20b6c0dc49a5247e096507ee01c970728bc/webrtc-c/canary/jobs/runner.groovy#L27-L42

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:42]: I think we should use `CANARY_CREDENTIALS` to be consistent with webrtc canary.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:80]: I think it's cleaner to use groovy to populate the env variables. For `PATH`, I would populate another env variable and append it to `PATH`, e.g. `export PATH=$CANARY_PATH:$PATH`.

Also, you don't need to separate `PATH` from `export PATH`, you can do this in one line, `export PATH=...`

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:12]: You can shorten this to just `mkdir -p build`. The `-p` will not create a new directory if it exists already.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:23]: ditto

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:123]: ditto, this can be removed and done in the runner wrapper

File: producer-c/producer-cloudwatch-integ/jobs/canary_seed.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/canary_seed.groovy]: Makes sense. Fixed everywhere

**Comment by disa6302** [comment]
Runs are failing after the changes. Will open up for review once it is fixed

File: consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/run-consumer.sh
Comment (lherman-cs) [inline consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/run-consumer.sh:1]: I'd highly recommend to use Makefile instead so we can even add this to our CI. 

To add this to the CI, you just need to create a Makefile that has "test" as the target, and add the directory path to the target in .travis.yaml below:

https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/e246e81370751088cb82b32be6e5176bb413d0f7/.travis.yml#L17

It's definitely not the best test, but it can add some sanity check at least.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:30]: Why not calling the build-consumer.sh script?

File: producer-c/producer-cloudwatch-integ/jobs/orchestrator.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/orchestrator.groovy:6]: this should be master

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:23]: Why do we need sudo? This scares me because this means that it can mutate the node's system.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [inline producer-c/producer-cloudwatch-integ/jobs/runner.groovy:77]: why do we need sudo?

File: consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/run-consumer.sh
Comment (lherman-cs) [reply consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/run-consumer.sh:1]: Webrtc Canary's Makefile, https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/master/webrtc-c/canary/Makefile

File: producer-c/producer-cloudwatch-integ/jobs/orchestrator.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/orchestrator.groovy:6]: It will change once the PR is approved.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/runner.groovy:23]: This is because the AWS CPP SDK related items get installed in root. Maybe I can install them in CMAKE_INSTALL_PREFIX instead and this sudo can go then.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/runner.groovy:30]: It is `run-consumer.sh`. I am adding it as a convenience to run locally. Here I am splitting the steps between `buildConsumer()` and `runConsumer().

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/runner.groovy:77]: Because the open-source directories have some files that cannot be removed without sudo. I know it is not the best, but this is the only way right now.

File: consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/run-consumer.sh
Comment (disa6302) [reply consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/run-consumer.sh:1]: Let me handle this in a seperate PR. As of now, run-consumer.sh is not being used in our Jenkins pipeline

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/runner.groovy]: Fixed after moving to CMAKE_INSTALL_PREFIX

File: producer-c/producer-cloudwatch-integ/jobs/orchestrator.groovy
Comment (disa6302) [reply producer-c/producer-cloudwatch-integ/jobs/orchestrator.groovy:6]: Will change it once PR is approved. I am running Jenkins with this branch right now.

File: producer-c/producer-cloudwatch-integ/jobs/runner.groovy
Comment (lherman-cs) [reply producer-c/producer-cloudwatch-integ/jobs/runner.groovy]: Oops sorry. I meant to delete this comment before posting it 😅

File: consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/Makefile
Comment (lherman-cs) [inline consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/Makefile:1]: Could you rename `consumer` to `test`? `test` is the target that the Travis script is looking for. Also, you just need to add a target to .travis.yaml to enable the CI, https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/833cbc117e206e09f7412731bf6b054adc53da4a/.travis.yml#L17

To:

```
- env: TARGET=webrtc-c/canary
- env: TARGET=consumer-java/aws-kinesis-video-producer-sdk-canary-consumer
```

**Review by lherman-cs — COMMENTED** [review]
Small comment about the CI integration. Also, could you squash your commit into a single commit?

**Comment by disa6302** [comment]
> Small comment about the CI integration. Also, could you squash your commit into a single commit?

Will do when I am about to merge.

File: consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/Makefile
Comment (disa6302) [reply consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/Makefile:1]: Sure!

**Review by lherman-cs — APPROVED** [review]
LGTM. Let's wait for the CI to pass.

**Review by lherman-cs — APPROVED** [review]

**Pull request merged**
